rqd/rqd/rqlogging.py

```diff
@@ -0,0 +1,149 @@
+#  Copyright Contributors to the OpenCue Project
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+"""Logging module, handles logging to files and non-files"""
+
+
+import logging
+import re
+import time
+import os
+import datetime
+import platform
+
+import rqd.rqconstants
+
+LOGTYPE_FILE = 1
+
+log = logging.getLogger(__name__)
+log.setLevel(rqd.rqconstants.CONSOLE_LOG_LEVEL)
+
+
+class RQDLogger(object):
+    """Class to abstract file logging, this class tries to act as a file object"""
+    filepath = None
+    fd = None
+    type = 0
+
+    def __init__(self, filepath):
+        """RQDLogger class initialization
+           @type    filepath: string
+           @param   filepath: The filepath to log to
+        """
+
+        protocolMatch = re.match(r"(?P<proto>[\w\-.:]+)://(?P<server>\w+)/(?P<filepath>.*)",
+                                 filepath)
```

Comment:
This is problema a problem for the Loki PR following this one, so for now my advice is to remove the LOGTYPE_FILE logic based on protocol matching from this PR. In my opinion, relying on the filepath format to define the protocol hides de functionality of the class on it's documentation instead of explicitly stating what formats are accepted.

One solution would be to change the constructor to `__init__(self, filepath=None, loki_path=None)` and require that at leas one of the paths is not None. But again, this can be handled on the Loki PR if we remove the LOGTYPE_FILE altogether from this PR as it doesn't make too much sense isolated.

Reply:
I agree, have removed the LOGTYPE_FILE